src/pages/storage/AttachDiskDeviceModal.tsx

```diff
@@ -0,0 +1,98 @@
+import { FC, KeyboardEvent, useState } from "react";
+import { Modal } from "@canonical/react-components";
+import FormLink from "components/FormLink";
+import BackLink from "components/BackLink";
+import { InstanceAndProfileFormikProps } from "components/forms/instanceAndProfileFormValues";
+import CustomVolumeModal from "./CustomVolumeModal";
+import { LxdStorageVolume } from "types/storage";
+import { LxdDiskDevice } from "types/device";
+import HostPathDeviceModal from "./HostPathDeviceModal";
+
+type DiskDeviceType = "custom volume" | "host path" | "choose type";
+
+interface Props {
+  close: () => void;
+  onFinish: (device: LxdStorageVolume | LxdDiskDevice) => void;
+  formik: InstanceAndProfileFormikProps;
+  project: string;
+}
+
+const AttachDiskDeviceModal: FC<Props> = ({
+  close,
+  formik,
+  project,
+  onFinish,
+}) => {
+  const [type, setType] = useState<DiskDeviceType>("choose type");
+
+  const handleEscKey = (e: KeyboardEvent<HTMLElement>) => {
+    if (e.key === "Escape") {
+      close();
+    }
+  };
+
+  const handleGoBack = () => {
+    setType("choose type");
+    return;
+  };
+
+  const modalTitle =
+    type === "choose type" ? (
+      "Choose disk type"
+    ) : (
+      <BackLink
+        title={`Attach ${type}`}
```

Comment:
I think this should say "Mount host path" in one case and "Attach custom volume" in the other.